crates/tauri-cli/src/interface/rust.rs

```diff
@@ -131,16 +131,18 @@ impl Interface for Rust {
         }
       })
       .unwrap();
-      watcher.watch(tauri_dir().join("Cargo.toml"), RecursiveMode::Recursive)?;
-      let (manifest, _modified) = rewrite_manifest(config)?;
-      let now = Instant::now();
-      let timeout = Duration::from_secs(2);
-      loop {
-        if now.elapsed() >= timeout {
-          break;
-        }
-        if rx.try_recv().is_ok() {
-          break;
+      watcher.watch(tauri_dir().join("Cargo.toml"), RecursiveMode::NonRecursive)?;
+      let (manifest, modified) = rewrite_manifest(config)?;
+      if modified {
+        let now = Instant::now();
+        let timeout = Duration::from_secs(2);
+        loop {
+          if now.elapsed() >= timeout {
+            break;
+          }
+          if rx.try_recv().is_ok() {
+            break;
+          }
         }
       }
```

Comment:
This is the changed part, other parts are just some small refactors that I did when looking through them

Also, I'm not entirely sure what does this watch even do, I would assume it's for waiting for the file modification to take effect from `rewrite_manifest`, is it?

Reply:
looks like it yeah. just so that it doesn't trigger a rebuild ig

Reply:
Added some comments and changed the loop to `recv_timeout`

Reply:
Whatever I guess, the timeout is just there to make sure we move on, if there're indeed an modification and no errors, the watcher should tell us before it, I'm not sure what time to set for the denouncer though, 1 sec feels too long to me, but I don't really care since it's rare, better safe then sorry I guess 🙃

Reply:
Your call 🤷